src/models/MyAccount.ts

```diff
@@ -0,0 +1,213 @@
+      "/holds?expand=record&fields=canFreeze,record,status,pickupLocation,frozen,patron,pickupByDate"
+    return await client.get(`${baseQuery}${holdsQuery}`)
+  }
+
+  static async fetchPatron(baseQuery: string) {
+    const patronQuery =
+      "?fields=names,barcodes,expirationDate,homeLibrary,emails,phones"
+    return await client.get(`${baseQuery}${patronQuery}`)
+  }
+
+  static async fetchFines(baseQuery: string) {
+    const finesQuery = "/fines"
+    return await client.get(`${baseQuery}${finesQuery}`)
+  }
+
+  static async fetchBibData(
+    holdsOrCheckouts: (SierraHold | SierraCheckout)[],
+    itemOrRecord: string
+  ) {
+    if (!holdsOrCheckouts.length) return []
+    const checkoutBibIds = holdsOrCheckouts.map((holdOrCheckout) => {
+      return holdOrCheckout[itemOrRecord].bibIds[0]
+    })
+
+    const defaultFields = await client.get(
+      `bibs?id=${checkoutBibIds}&fields=default,varFields`
+    )
+
+    return defaultFields.entries.reduce(
+      (
+        bibDataMap: Record<
+          string,
+          { title: string; isResearch: boolean; isNyplOwned: boolean }
+        >,
+        bibFields
+      ) => {
+        let isResearch
+        let isNyplOwned
+        const title = bibFields.title
+        const nineTen = bibFields.varFields.find(
+          (field) => field.marcTag === "910"
+        )
+        // if we are unsure of the research ness of a bib, default to true so
+        // we don't let them renew or freeze the record
+        if (!nineTen) {
+          isResearch = true
+          isNyplOwned = false
+        } else {
+          const nineTenContent = nineTen.subfields.find(
+            (subfield: { tag: string; subfield: string }) =>
+              subfield.tag === "a"
+          ).content
+          isResearch = nineTenContent.startsWith("RL")
+          isNyplOwned = nineTenContent !== "RLOTF"
+        }
+        bibDataMap[bibFields.id] = { title, isResearch, isNyplOwned }
+        return bibDataMap
+      },
+      {}
+    )
+  }
+
+  buildHolds(holds: SierraHold[], bibDataMap): Hold[] {
+    return holds.map((hold: SierraHold) => {
+      return {
+        patron: MyAccount.getRecordId(hold.patron),
+        id: MyAccount.getRecordId(hold.id),
+        pickupByDate: hold.pickupByDate,
+        canFreeze: hold.canFreeze,
+        frozen: hold.frozen,
+        status: MyAccount.getStatus(hold.status),
+        pickupLocation: hold.pickupLocation.name,
+        title: bibDataMap[hold.record.bibIds[0]].title,
+        isResearch: bibDataMap[hold.record.bibIds[0]].isResearch,
+        bibId: hold.record.bibIds[0],
+        isNyplOwned: bibDataMap[hold.record.bibIds[0]].isResearch,
+      }
+    })
+  }
+
+  buildPatron(patron: SierraPatron): Patron {
+    return {
+      name: patron.names[0],
+      barcode: patron.barcodes[0],
+      expirationDate: patron.expirationDate,
+      primaryEmail: patron.emails[0],
+      emails: patron.emails,
+      primaryPhone: patron.phones[0].number,
```

Comment:
If `patron.phones` is an empty array, this will fail.

Reply:
Even if a phone number is required by the ILS for a patron account, let's assume it's not to be on the safer side.